src/main/scala/viper/gobra/translator/encodings/FloatEncoding.scala

```diff
@@ -0,0 +1,297 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2021 ETH Zurich.
+
+package viper.gobra.translator.encodings
+
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.theory.Addressability.{Exclusive, Shared}
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.translator.util.ViperWriter.CodeLevel._
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class FloatEncoding extends LeafTypeEncoding {
+
+  import viper.gobra.translator.util.TypePatterns._
+
+  /**
+    * Translates a type into a Viper type.
+    */
+  override def typ(ctx: Context): in.Type ==> vpr.Type = {
+    case ctx.Float32() / Exclusive => floatType32
+    case ctx.Float64() / Exclusive => floatType64
+    case ctx.Float32() / Shared => vpr.Ref
+    case ctx.Float64() / Shared => vpr.Ref
+  }
+
+  private lazy val floatType32: vpr.Type = {
+    isUsed32 = true
+    vpr.Int
+  }
+
+  private lazy val floatType64: vpr.Type = {
+    isUsed64 = true
+    vpr.Int
+  }
+
+  /**
+    * Encodes expressions as values that do not occupy some identifiable location in memory.
+    * X stands for either 32 or 64 below:
+    * [ dflt(x: floatX) ] -> defaultValueX()
+    * [ (x: floatX) + (y: floatX) ] -> addFloatX([ x ], [ y ])
+    * [ (x: floatX) - (y: floatX) ] -> subFloatX([ x ], [ y ])
+    * [ (x: floatX) * (y: floatX) ] -> mulFloatX([ x ], [ y ])
+    * [ (x: floatX) / (y: floatX) ] -> divFloatX([ x ], [ y ])
+    * [ int(x: floatX) ] -> fromIntToX([ x ])
+    */
+  override def expr(ctx: Context): in.Expr ==> CodeWriter[vpr.Exp] = {
+
+    def goE(x: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(x)(ctx)
+
+    default(super.expr(ctx)) {
+      case (e: in.DfltVal) :: ctx.Float32() / Exclusive =>
+        unit(withSrc(vpr.FuncApp(func = defaultValue32, Seq()), e))
+      case (e: in.DfltVal) :: ctx.Float64() / Exclusive =>
+        unit(withSrc(vpr.FuncApp(func = defaultValue64, Seq()), e))
```

Comment:
You could make the default value the conversion from 0 to the respective type (e.g. `float32(0)` ).